src/Chapter04.Tests/Listing04.28.Tests.cs

```diff
@@ -1,15 +1,15 @@
+using AddisonWesley.Michaelis.EssentialCSharp.Shared.Tests;
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter04.Listing04_28.Tests;
 
 [TestClass]
 public class ProgramTests
 {
+    private static readonly string[] ExpectedErrorIds = new string[] { "CS0103" };
```

Comment:
nit: field is not named with an underscore.
```suggestion
    private static readonly string[] _ExpectedErrorIds = new string[] { "CS0103" };
```